pyccel/codegen/printing/pycode.py

```diff
@@ -442,6 +442,12 @@ def _print_PythonList(self, expr):
         args = ', '.join(self._print(i) for i in expr.args)
         return '['+args+']'
 
+    def _print_PythonSet(self, expr):
+        args = ', '.join(self._print(i) for i in expr.args)
+        if len(expr.args) == 0:
+            return 'set()'
+        return '{'+args+'}'
```

Comment:
Should you not place the `if` condition before computing the `args` string?

Reply:
Sure, could you please tell me why?

Reply:
Because there is no need to compute `args` if the length of `expr.args` is 0.

Reply:
fixed : 2ed79da